mmv1/templates/terraform/update_encoder/reservation.go.erb

```
@@ -11,8 +11,107 @@
+		transformed["shareType"] = "SPECIFIC_PROJECTS"
+		// Set project_map.
+		projectMap := make(map[string]interface{})
+		old, new := d.GetChange("share_settings")
+		oldMap := old.([]interface{})[0].(map[string]interface{})["project_map"]
+		newMap := new.([]interface{})[0].(map[string]interface{})["project_map"]
+		before := oldMap.(*schema.Set)
+		after := newMap.(*schema.Set)
+
+		for _, raw := range after.Difference(before).List() {
+			original := raw.(map[string]interface{})
+			singleProject := make(map[string]interface{})
+			// set up project_map.
+			transformedProjectId := original["project_id"]
+			if val := reflect.ValueOf(transformedProjectId); val.IsValid() && !isEmptyValue(val) {
+				singleProject["projectId"] = transformedProjectId
+			}
+			transformedId, err := expandString(original["id"], d, config)
+			if err != nil {
+				return newObj, nil
+			}
+			projectMap[transformedId] = singleProject
+			// add added projects to updateMask
+			if firstProject != true {
+				maskId = fmt.Sprintf("%s%s", "&paths=shareSettings.projectMap.", original["project_id"])
+			} else {
+				maskId = fmt.Sprintf("%s%s", "?paths=shareSettings.projectMap.", original["project_id"])
+				firstProject = false
+			}
+			decodedPath, _ := url.QueryUnescape(maskId)
+			urlUpdateMask = urlUpdateMask + decodedPath
+		}
+		transformed["projectMap"] = projectMap
+		newObj["shareSettings"] = transformed
+
+		// add removed projects to updateMask
+		firstProject = true
+		for _, raw := range before.Difference(after).List() {
+			original := raw.(map[string]interface{})
+			// To remove a project we need project number.
+			projectId := fmt.Sprintf("%s", original["project_id"])
+			projectIdOrNum := projectId
+			_, err := strconv.Atoi(projectId)
+			// convert id to number.
+			if err != nil {
+				config := meta.(*Config)
+				project, err := config.NewResourceManagerClient(config.userAgent).Projects.Get(projectId).Do()
+				if err != nil {
+					return newObj, nil
```

Should we return the error in this case, rather than silently return the partial new `obj`?

---

thanks for the point, same applied for the other returns.